### ISSUE TITLE
Yoshi does not support globbing

### ISSUE THREAD

**Issue by rokasmik** [wix/yoshi#480]
# 🐛 Bug Report

### 😯 Current Behavior

`yoshi lint src/**/* test/**/*` just lints the second level and does not lint all the files inside the folder.

### 🤔 Expected Behavior

`yoshi lint src/**/*` should lint all the files inside the src folder

### 🔦 Context

Yoshi 3 lints my dynamically generated files, as they are included in tsconfig for use in the project. I can't exclude it the [--file] option as it does not support globbing. I can't migrate to yoshi 3 with prettier as prettier finds a lot of errors in the generated files.

### 💁 Possible Solution

Support globbing in files pattern so I could specify what I want to lint

### 👣 Steps to Reproduce

install yoshi and try linting specific directories

### 🌍 Your Environment
yoshi 3, mac os x

**Comment by ranyitz** [comment]
@rokasmik 

#211 Was intended for `lint-stage` support.

Why do you want to change the default glob patterns of yoshi?

**Comment by ranyitz** [comment]
@rokasmik 
So, eslint supports `.eslintignore` [for ignoring files and directories](https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories) out of the box. You can also use `tslint.json` with the [exclude option](https://palantir.github.io/tslint/usage/configuration/). Let me know if it's solves your problem.

**Comment by rokasmik** [comment]
Thank you, I achieved ignoring `*.d.ts` files with `linterOptions` & `exclude`, but couldn't get `exclude` to respect js files, so I also needed to add this to `tslint.json`:
```
  "jsRules": {
    "prettier": false
  },
```
This works for my case, but wouldn't be sufficient if i had "allowJs" option on and would want to ignore some js files but lint other.

**Comment by rokasmik** [comment]
@ranyitz so the solution I described above does not really solve the issue, it is too hacky. I still can't find a way to exclude a js file from tslint. `exclude` does not care

**Comment by ranyitz** [comment]
@rokasmik I see, can you please send a link to your repo including "steps to reproduce"?

**Comment by rokasmik** [comment]
@ranyitz sure, will contact you with PM

**Comment by rokasmik** [comment]
Sharing it here:
So I have this PR hanging in my repo, using `tslint-config-yoshi` instead of `tslint-config-wix`
https://github.com/wix-private/app-market/pull/2862
repo `app-market`, project `membership/membership-tpa`, branch `paid-plan-lint`
Steps:
`cd app-market/membership/membership-tpa && npm i && npm run build:proto && npm run lint`
So now I managed the get the build passing with manually adding `//tslint:disable` to the generated file, but the linting takes a lot of time as it still goes through the generated file which is like 30k lines long. So that doesn’t work for us, we use lint in the precommit hook.
So I only need to lint src and test categories. Tried all kind of stuff with tsconfig.json, but excluding the generated js file (membership-tpa/generated/service-api.js) will exclude it from the build and it won’t work (npm run test will fail).
I also have it excluded in `tslint.json` - linterOptions `"exclude": ["generated/service-api.d.ts", "generated/service-api.js"]`. It doesn’t work 😕
